src/gui/editors/PianoRoll.cpp

```diff
@@ -804,15 +818,15 @@ void PianoRoll::drawNoteRect(QPainter & p, int x, int y,
 
 	p.setPen( col );
 	p.setRenderHint(QPainter::Antialiasing);
-	p.drawRoundedRect( x, y, width, KEY_LINE_HEIGHT-1, 5, 2 );
+	p.drawRoundedRect( QRectF ( x + 0.5, y - 0.5, width, KEY_LINE_HEIGHT ), radiusX, radiusY );
 
 	// that little tab thing on the end hinting at the user
 	// to resize the note
 	p.setPen( noteCol.lighter( 200 ) );
+	p.setBrush( noteCol.lighter( 200 ) );
 	if( width > 2 )
 	{
-		p.drawLine( x + width - 3, y + 3, x + width - 3,
-						y + KEY_LINE_HEIGHT - 5 );
+		p.drawRect( QRectF (x + width - 2.5, y + 3.5, 1, KEY_LINE_HEIGHT - 8 ) );
```

Comment:
I think there are too many "magic constants" on this line. How do you feel about changing it to something like:
```c++
int leftIndent = 2.5;
int vertIndent = 3.5;
p.drawRect( QRectF (x + width - leftIndent, y + vertIndent, 1, KEY_LINE_HEIGHT - 2*vertIndent) );
```
It's not pretty, but it does add some much-needed context to the numbers & I don't see a better way to do that.

Similarly, is there an off-by-one error here? It seems the line's distance from the top edge is 3.5, whereas the distance from the bottom edge is (8 - 3.5) = 4.5 pixels, unless I'm misunderstanding this statement.